src/pks/flow/overland_pressure_ti.cc

```diff
@@ -293,183 +294,37 @@ void OverlandPressureFlow::UpdatePreconditioner(double t, Teuchos::RCP<const Tre
   //       Acc_cells[c] -= cv[0][c] * dq_dp[0][c] / dh_dp[0][c];
   //     }
   //   }
-  // }      
-  
+  // }
+
 
   // 3. Assemble and precompute the Schur complement for inversion.
   // 3.a: Patch up BCs in the case of zero conductivity
   FixBCsForPrecon_(S_next_.ptr());
 
-  // // 3.c: Add in full Jacobian terms
-  // if (tpfa_ && full_jacobian_) {
-  //   if (vo_->os_OK(Teuchos::VERB_EXTREME))
-  //     *vo_->os() << "    including full Jacobian terms" << std::endl;
-
-  //   Teuchos::RCP<const CompositeVector> depth =
-  //       S_next_->GetFieldData("ponded_depth");
-  //   Teuchos::RCP<const CompositeVector> pres_elev =
-  //       S_next_->GetFieldData("pres_elev");
-
-  //   // Update conductivity.  Note the change of variables from pressure to
-  //   // height.
-  //   // -- Krel_face gets conductivity
-  //   S_next_->GetFieldEvaluator("overland_conductivity")
-  //       ->HasFieldDerivativeChanged(S_next_.ptr(), name_, key_);
-  //   Teuchos::RCP<const CompositeVector> cond =
-  //       S_next_->GetFieldData("overland_conductivity");
-  //   Teuchos::RCP<const CompositeVector> dcond_dp =
-  //       S_next_->GetFieldData("doverland_conductivity_dponded_depth");
-  //   CompositeVector dcond_dh(*dcond_dp);
-  //   dcond_dh.ViewComponent("cell",false)->ReciprocalMultiply(1., dh_dp,
-  //           *dcond_dp->ViewComponent("cell",false), 0.);
-
-  //   // -- Add in the Jacobian
-  //   tpfa_preconditioner_->AnalyticJacobian(*upwinding_,
-  //                                          S_next_.ptr(), "pres_elev",
-  //                                          dcond_dh, bc_markers_,
-  //                                          bc_values_);
-  // }
-
   preconditioner_diff_->ApplyBCs(true, true, true);
-  
+
   // 3.d: Rescale to use as a pressure matrix if used in a coupler
   //  if (coupled_to_subsurface_via_head_ || coupled_to_subsurface_via_flux_) {
   if (!precon_used_) {
     // Scale Spp by dh/dp (h, NOT h_bar), clobbering rows with p < p_atm
-    std::string pd_key = Keys::getKey(domain_,"ponded_depth");
+    std::string pd_key = pd_key_;
```

Comment:
Is this copy needed? we can use pd_key_ directly

Reply:
Yeah, leftovers from refactor-by-sed...